foreman.te

```diff
@@ -501,34 +481,183 @@ typealias bin_t alias foreman_hook_t;
+    logging_list_logs(passenger_t)
+    write_files_pattern(passenger_t, foreman_log_t, foreman_log_t)
+    allow passenger_t foreman_log_t:dir { create_file_perms append_file_perms list_dir_perms search_dir_perms };
+    allow passenger_t foreman_log_t:file { create_file_perms append_file_perms };
+    corenet_tcp_connect_all_ports(passenger_t)
+    files_read_usr_files(passenger_t)
+    term_search_ptys(passenger_t)
+    files_exec_usr_files(passenger_t)
+    dev_read_sysfs(passenger_t)
+    dev_search_sysfs(passenger_t)
+    dev_read_rand(passenger_t)
+    allow passenger_t self:process setsockcreate;
+    miscfiles_manage_cert_dirs(passenger_t)
+    miscfiles_manage_generic_cert_files(passenger_t)
+    can_exec(passenger_t, passenger_tmp_t)
+    admin_pattern(httpd_t, passenger_tmp_t, passenger_tmp_t)
+    admin_pattern(passenger_t, httpd_tmp_t, httpd_tmp_t)
+    corenet_tcp_connect_smtp_port(passenger_t)
+')
+optional_policy(`
+    tunable_policy(`passenger_run_foreman', `
+        postgresql_stream_connect(passenger_t)
+    ')
+')
+optional_policy(`
+    tunable_policy(`passenger_run_foreman', `
+        admin_pattern(httpd_t, foreman_lib_t, foreman_lib_t)
+        admin_pattern(passenger_t, foreman_lib_t, foreman_lib_t)
+        admin_pattern(passenger_t, foreman_var_run_t, foreman_var_run_t)
+        corenet_tcp_connect_virt_port(passenger_t)
+        corenet_tcp_connect_ssh_port(passenger_t)
+        read_files_pattern(passenger_t, foreman_db_t, foreman_db_t)
+        write_files_pattern(passenger_t, foreman_db_t, foreman_db_t)
+        corenet_udp_bind_generic_port(passenger_t)
+        admin_pattern(passenger_t, passenger_tmp_t, passenger_tmp_t)
+    ')
+')
+optional_policy(`
+    tunable_policy(`passenger_run_foreman', `
+        allow passenger_t self:process getsession;
+        fs_rw_anon_inodefs_files(passenger_t)
+        allow passenger_t httpd_t:unix_stream_socket { read write getattr };
+        fs_getattr_xattr_fs(passenger_t)
+        allow passenger_t self:capability2 block_suspend;
+    ')
+')
+optional_policy(`
+    tunable_policy(`passenger_run_foreman', `
+        corenet_tcp_connect_puppet_port(httpd_t)
+        corenet_tcp_connect_puppet_port(passenger_t)
+    ')
+')
+optional_policy(`
+    tunable_policy(`passenger_run_foreman', `
+        corenet_tcp_connect_http_port(httpd_t)
+        corenet_tcp_connect_http_port(passenger_t)
+    ')
+')
+optional_policy(`
+    tunable_policy(`passenger_run_foreman', `
+        mta_send_mail(passenger_t)
+    ')
+')
+optional_policy(`
+    tunable_policy(`passenger_run_foreman', `
+        mysql_stream_connect(passenger_t)
+        mysql_list_db(passenger_t)
+    ')
+')
+optional_policy(`
+    tunable_policy(`passenger_run_foreman', `
+        hostname_exec(passenger_t)
+    ')
+')
+optional_policy(`
+    tunable_policy(`passenger_run_foreman', `
+        corecmd_search_bin(passenger_t)
+        domtrans_pattern(passenger_t, puppetmaster_exec_t, puppetmaster_t)
+    ')
+')
```

Comment:
Foreman never talks to the Puppetmaster

Reply:
Well, we do carry whole policy for puppetmaster as it also runs in passenger (the old version). I am keeping these until passenger is completely removed. Granted probably all of our users uses the Java passenger.

This rulle is a domain transition for puppet master (the old version). If I'd remove this, I'd need to remove pretty much all of the legacy passenger stuff too, which is planned in a separate PR.